apps/src/templates/progress/LessonGroup.jsx

```diff
@@ -107,13 +111,19 @@ class LessonGroup extends React.Component {
       lessons,
       levelsByLesson,
       isSummaryView,
-      isPlc
+      isPlc,
+      lessonIsVisible
     } = this.props;
 
     const TableType = isSummaryView
       ? SummaryProgressTable
       : DetailProgressTable;
-    const icon = this.state.collapsed ? 'caret-right' : 'caret-down';
+
+    const hasVisibleLesson = lessons.some(lesson => lessonIsVisible(lesson));
+
+    if (!hasVisibleLesson) {
```

Comment:
Do we want to hide the lesson group for both students and teachers or just for students?

Reply:
So when we talked yesterday I thought that summaryView was related to the student or teacher view since it was being toggled when I changed views, but actually it's just the two different views of the lessons. The `lessonIsVisible` function takes into account the `viewType` and shows hidden lessons to teachers https://github.com/code-dot-org/code-dot-org/blob/staging-next/apps/src/templates/progress/progressHelpers.js#L34